Sources/WalletConnectPush/Client/Wallet/SubscriptionScopeProvider.swift

```diff
@@ -0,0 +1,22 @@
+
+import Foundation
+
+class SubscriptionScopeProvider {
+    enum Errors: Error {
+        case invalidUrl
+    }
+
+    private var cache = [String: Set<NotificationType>]()
```

Comment:
Could it be persistent?

Reply:
I prefer not to because it may change od dapp's domain and and cache would not reflect the actual config. 
And persistence also would not bring much value as you probably will subscribe not very often.